src/ast/passes/codegen_llvm.cpp

```diff
@@ -134,14 +134,12 @@ void CodegenLLVM::visit(PositionalParameter &param)
           expr_ = b_.getInt64(std::stoull(pstr, nullptr, 0));
         }
       } else {
-        Constant *const_str = ConstantDataArray::getString(
-            module_->getContext(), pstr, true);
-        AllocaInst *buf = b_.CreateAllocaBPF(
-            ArrayType::get(b_.getInt8Ty(), pstr.length() + 1), "str");
-        b_.CreateMemsetBPF(buf, b_.getInt8(0), pstr.length() + 1);
-        b_.CreateStore(const_str, buf);
-        expr_ = b_.CreatePtrToInt(buf, b_.getInt64Ty());
-        expr_deleter_ = [this, buf]() { b_.CreateLifetimeEnd(buf); };
+        auto string_param = llvm::dyn_cast<GlobalVariable>(
+            module_->getOrInsertGlobal(
+                pstr, ArrayType::get(b_.getInt8Ty(), pstr.length() + 1)));
```

Comment:
thoughts on a more generic name of the global var other than the string value itself? the contents are already visible on the same line in IR

could use something like "const_string" and let llvm generate disambiguators

Reply:
What are the benefits of a more generic name? At least when the global var is used later, you also see the string e.g.
```
call void @llvm.memcpy.p0.p0.i64(ptr align 1 %2, ptr align 1 @abc, i64 4, i1 false)
```

Reply:
Yeah good point. As is makes sense